Extensions/alternate_audio_player.js

```diff
@@ -0,0 +1,60 @@
+//* TITLE Alternate Audio Player **//
+//* VERSION 0.1.0 **//
+//* DESCRIPTION	Adds an alternative (working) audio player to audio posts. **//
+//* DETAILS This extension adds a native HTML5 audio player to every audio post, which should always work as well as enabling looping, downloading, and better seeking. **//
+//* DEVELOPER akunohomu **//
+//* FRAME false **//
+//* BETA true **//
+
+XKit.extensions.alternate_audio_player = new Object({
+
+	running: false,
+
+	preferences: {
+		"default_volume": {
+			text: "Default volume",
+			default: 1.0,
+			value: 1.0,
+			type: "combo",
+			values: [
+				"100%", 1.0,
+				"75%", 0.75,
+				"50%", 0.50,
+				"25%", 0.25
+			],
+			desktop_only: true
+		}
+	},
+
+	run: function() {
+		"use strict";
+		this.running = true;
+		XKit.post_listener.add("alternate_audio_player", this.add_audio);
+		this.add_audio();
+	},
+
+	add_audio: function() {
+		"use strict";
+		jQuery(".audio_player_container").not(".xkit-audio-added").each(function(index) {
+			var ob = jQuery(this);
+			ob.addClass("xkit-audio-added");
+			var url = ob.attr("data-stream-url");
+			var key = ob.attr("data-post-key");
+			if (typeof url !== "undefined" && typeof key !== "undefined") {
+				var player = jQuery("<audio class='xkit-audio-player' src='" + url + "?play_key=" + key +
```

Comment:
I'm uncertain how to do that (this is by *far* the most elaborate thing I've written in JavaScript) but as far as I can tell, `data-post-key` comes as a pre-escaped group of URL parameters. The HTML generated by this looks correct.